src/WebJobs.Script.WebHost/Configuration/ScriptApplicationHostOptionsSetup.cs

```diff
@@ -30,6 +30,9 @@ public void Configure(ScriptApplicationHostOptions options)
             _configuration.GetSection(ConfigurationSectionNames.WebHost)
                     ?.Bind(options);
 
+            // Indicate that a WebHost is hosting the ScriptHost
+            options.HasParentScope = true;
```

Comment:
This is meant to indicate to the child JobHost that it's running inside of a parent. With that information, it'll skip setting up its common services, as the parent will have them set.